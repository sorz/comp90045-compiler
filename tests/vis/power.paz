program power;

var x: real;
    result: real;
    n: integer;

procedure power(x: real; n: integer; var out: real);
    var res: real;
begin
    if n = 1 then
        out := x
    else
    begin
        power(x*x, n div 2, res);
        if 2*(n div 2) = n then
            out := res          { n was even }
        else
            out := x * res      { n was odd }
    end
end;

begin
    write('Float x: ');
    read(x);
    write('Positive integer n: ');
    read(n);
    power(x, n, result);
    write('x^n is: ');
    write(result);
    writeln
end.
