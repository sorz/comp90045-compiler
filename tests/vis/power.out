Float x: Positive integer n: x^n is: 125.000000
